src/evict/evict_lru.c

```diff
@@ -2904,13 +2890,18 @@ __wti_evict_app_assist_worker(WT_SESSION_IMPL *session, bool busy, bool readonly
         if (!busy && __wt_atomic_loadv64(&txn_shared->pinned_id) != WT_TXN_NONE &&
           __wt_atomic_loadv64(&txn_global->current) != __wt_atomic_loadv64(&txn_global->oldest_id))
             busy = true;
-        max_progress = busy ? 5 : 20;
+        uint64_t max_progress = busy ? 5 : 20;
 
         /* See if eviction is still needed. */
         if (!__wt_evict_needed(session, busy, readonly, &pct_full) ||
           (pct_full < 100.0 &&
             (__wt_atomic_loadv64(&evict->eviction_progress) > initial_progress + max_progress)))
-            break;
+            goto err;
+
+        if (!__evict_check_user_ok_with_eviction(session, busy)) {
+            ret = WT_ERROR; /* Prevent from requesting a rollback in the err handler. */
```

Comment:
I'm not familiar with what is going on here. Why are preventing from request a rollback in the error handler here?

Reply:
Hmm actually you're right, it shouldn't be done here!